src/util/streamService.js

```diff
@@ -0,0 +1,78 @@
+import { connect } from 'getstream';
+
+/**
+ * A class that abstracts calls to the GetStream.io API.
+ *
+ * @class StreamService
+ */
+class StreamService {
+	/**
+	 * Creates an instance of StreamService.
+	 *
+	 * @param {string} apiKey - The GetStream.io API key.
+	 * @param {string} appId - The GetStream.io App id.
+	 * @param {string} authToken - The auth user token.
+	 * @memberof StreamService
+	 */
+	constructor(apiKey, authToken, appId) {
+		/**
+		 * The GetStream.io client instance.
+		 *
+		 * @type {Stream.Client}
+		 */
+		this.client = connect(apiKey, authToken, appId);
+	}
+
+	/**
+	 * Creates a user feed for the specified user ID.
+	 *
+	 * @param {string} userId - The ID of the user.
+	 * @returns {Stream.Feed}
+	 * @memberof StreamService
+	 */
+	createUserFeed(userId) {
+		return this.client.feed('user', userId);
+	}
+
+	/**
+	 * Follows another user.
+	 *
+	 * @param {string} sourceUserId - The ID of the user initiating the follow.
+	 * @param {string} targetUserId - The ID of the user being followed.
+	 * @returns {Promise<void>}
+	 * @memberof StreamService
+	 */
+	followUser(sourceUserId, targetUserId) {
+		const sourceFeed = this.createUserFeed(sourceUserId);
+		const targetFeed = this.createUserFeed(targetUserId);
+
+		return sourceFeed.follow(targetFeed);
+	}
+
+	/**
+	 * Adds an activity to the specified feed.
+	 *
+	 * @param {Stream.Feed} feed - The feed to which the activity will be added.
+	 * @param {object} activity - The activity object to add
+	 * Read: https://getstream.io/activity-feeds/docs/javascript/adding_activities/?language=javascript#fields
+	 * @returns {Promise<void>}
+	 * @memberof StreamService
+	 */
+	static addActivity(feed, activity) {
```

Comment:
What is the shape of the `activity` object? This class should probably own what that means, so splitting this out into params would probably make sense.

OR, if we move this to CPS (see other comment), then we can use TypeScript for type safety here.